### PR TITLE
fix: JDBCTOGCS add input query as argument

### PR DESCRIPTION
**Pull request by tanyarw**

### CODE REVIEW

**Comment by surjits254** [comment]
Please update readme as well

File: python/dataproc_templates/jdbc/README.md [hunk 1]
```diff
@@ -334,6 +335,7 @@ usage: main.py --template JDBCTOGCS \
     --jdbctogcs.input.url JDBCTOGCS.INPUT.URL \
     --jdbctogcs.input.driver JDBCTOGCS.INPUT.DRIVER \
     --jdbctogcs.input.table JDBCTOGCS.INPUT.TABLE \
+    --jdbctogcs.input.sql.query JDBCTOGCS.INPUT.SQL.QUERY \ (refer NOTE)
```
Comment (tanyarw) [inline python/dataproc_templates/jdbc/README.md:338]: I wasn't sure how to document this, let me know if there could be a better way!

Comment (vanshaj-bhatia) [reply python/dataproc_templates/jdbc/README.md:338]: Lets remove either one of them from usage as they would never be used together. I would suggest adding note below for the usage for jdbctogcs.input.sql.query. ref. https://github.com/GoogleCloudPlatform/dataproc-templates/tree/main/java/src/main/java/com/google/cloud/dataproc/templates/jdbc#:~:text=username%3E%26password%3D%3Cpassword%3E%27-,Have%20SQL%20query%20within%20double%20quotes.%20Example%2C,-%2D%2DtemplateProperty%20%20%27jdbctogcs.sql

File: python/dataproc_templates/jdbc/README.md [hunk 2]
```diff
@@ -362,6 +364,7 @@ export JARS="<gcs_path_to_jdbc_jar_files>/mysql-connector-java-8.0.29.jar,<gcs_p
 --jdbctogcs.input.url="jdbc:mysql://<hostname>:<port>/<dbname>?user=<username>&password=<password>" \
 --jdbctogcs.input.driver=<jdbc-driver-class-name> \
 --jdbctogcs.input.table=<input table name or subquery with where clause filter> \
+--jdbctogcs.input.sql.query=<input sql query> \ (refer NOTE)
```
Comment (vanshaj-bhatia) [inline python/dataproc_templates/jdbc/README.md:367]: Lets remove jdbctogcs.input.sql.query from here, as both would never be used together. Below note justifies the usage correctly.

File: python/dataproc_templates/jdbc/jdbc_to_gcs.py [hunk 3]
```diff
@@ -148,10 +154,10 @@ def parse_args(args: Optional[Sequence[str]] = None) -> Dict[str, Any]:
         known_args: argparse.Namespace
         known_args, _ = parser.parse_known_args(args)
 
-        if getattr(known_args, constants.JDBCTOGCS_INPUT_TABLE) and getattr(known_args, constants.JDBCTOGCS_SQL_QUERY):
+        if getattr(known_args, constants.JDBCTOGCS_INPUT_TABLE) and getattr(known_args, constants.JDBCTOGCS_INPUT_SQL_QUERY):
             sys.exit('ArgumentParser Error: Arguments cannot have both input table and sql query, use either one.')
 
-        if getattr(known_args, constants.JDBCTOGCS_SQL_QUERY) and not getattr(known_args, constants.JDBCTOGCS_TEMP_VIEW_NAME):
+        if getattr(known_args, constants.JDBCTOGCS_TEMP_SQL_QUERY) and not getattr(known_args, constants.JDBCTOGCS_TEMP_VIEW_NAME):
```
Comment (vanshaj-bhatia) [inline python/dataproc_templates/jdbc/jdbc_to_gcs.py:160]: What will happen if temp view is given but temp query is empty?

File: python/dataproc_templates/jdbc/jdbc_to_gcs.py [hunk 4]
```diff
@@ -188,33 +195,60 @@ def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
...
                 .option(constants.JDBC_DRIVER, input_jdbc_driver) \
-                .option(constants.JDBC_TABLE, input_jdbc_table) \
+                .option(constants.JDBC_QUERY, input_jdbc_sql_query) \
                 .option(constants.JDBC_PARTITIONCOLUMN, input_jdbc_partitioncolumn) \
                 .option(constants.JDBC_LOWERBOUND, input_jdbc_lowerbound) \
                 .option(constants.JDBC_UPPERBOUND, input_jdbc_upperbound) \
                 .option(constants.JDBC_NUMPARTITIONS, jdbc_numpartitions) \
                 .option(constants.JDBC_FETCHSIZE, input_jdbc_fetchsize) \
                 .load()
+        else:
```
Comment (vanshaj-bhatia) [inline python/dataproc_templates/jdbc/jdbc_to_gcs.py:244]: Can this be done neatly? May be using .options with a dict and adding properties that are required as per single level if-else statements.

**Review by vanshaj-bhatia — CHANGES_REQUESTED** [review]

File: python/dataproc_templates/jdbc/jdbc_to_gcs.py (hunk 3 above)
Comment (tanyarw) [reply python/dataproc_templates/jdbc/jdbc_to_gcs.py:160]: The temp view will be created only if temp query is given. `if temp_sql_query : ` -> create temp view & execute query

File: python/dataproc_templates/jdbc/jdbc_to_gcs.py (hunk 4 above)
Comment (tanyarw) [reply python/dataproc_templates/jdbc/jdbc_to_gcs.py]: made changes

**Review by vanshaj-bhatia — APPROVED** [review]
LGTM!
Please test the template once before merging.

**Pull request merged**